code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -284,7 +311,266 @@
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(QDELETED(A))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+	var/mob/living/C = A
+
+	if(!isliving(C))
+		to_chat(X, "<span class='xenowarning'>We cannot psychically mark this target!</span>")
+		return FALSE
+
+	if(C == X)
+		to_chat(X, "<span class='xenowarning'>Why would we target ourselves?</span>")
+		return FALSE
+
+	if(X.on_fire)
+		to_chat(X, "<span class='xenowarning'>We're too busy being on fire to mark them!</span>")
+		return FALSE
+
+	if(!X.line_of_sight(C)) //Need line of sight.
+		to_chat(X, "<span class='xenowarning'>We require line of sight to mark them!</span>")
+		return FALSE
+
+	return TRUE
+
+
+/datum/action/xeno_action/activable/hunter_mark/on_cooldown_finish()
+	var/mob/living/carbon/xenomorph/X = owner
+	to_chat(X, "<span class='xenowarning'><b>We are able to impose our psychic mark again.</b></span>")
+	X.playsound_local(X, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+
+/datum/action/xeno_action/activable/hunter_mark/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/M = A
+
+	if(!isliving(M)) //Sanity
+		to_chat(X, "<span class='xenodanger'>We cannot mark this target!</span>")
+		fail_activate()
+
+	X.face_atom(M) //Face towards the target so we don't look silly
+
+	var/name = M.name
+
+	to_chat(X, "<span class='xenodanger'>We prepare to psychically mark [M.name] as our quarry.</span>")
+
+	if(!do_after(X, HUNTER_MARK_WINDUP, TRUE, target, BUSY_ICON_HOSTILE)) //Slight wind up
+		return fail_activate()
+
+	if(!X.line_of_sight(M)) //Need line of sight.
+		to_chat(X, "<span class='xenowarning'>We lost line of sight to the target!</span>")
+		return fail_activate()
+
+	X.hunter_mark_target = M //Set our target
+	RegisterSignal(X.hunter_mark_target, COMSIG_PARENT_PREQDELETED, .proc/unset_target) //For var clean up
+
+	to_chat(X, "<span class='xenodanger'>We psychically mark [name] as our quarry.</span>")
+	X.playsound_local(X, 'sound/effects/ghost.ogg', 25, 0, 1)
+
+	succeed_activate()
+
+	GLOB.round_statistics.hunter_marks++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hunter_marks") //Statistics
+	add_cooldown()
+
+/datum/action/xeno_action/activable/hunter_mark/proc/unset_target()
+	var/mob/living/carbon/xenomorph/X = owner
+	UnregisterSignal(X.hunter_mark_target, COMSIG_PARENT_PREQDELETED)
+	X.hunter_mark_target = null
+
+// ***************************************
+// *********** Psychic Trace
+// ***************************************
+/datum/action/xeno_action/psychic_trace
+	name = "Psychic Trace"
+	action_icon_state = "toggle_queen_zoom"
+	mechanics_text = "Psychically ping the creature you marked, letting you know its direction, distance and location, and general condition."
+	plasma_cost = 1 //Token amount
+	keybind_signal = COMSIG_XENOABILITY_PSYCHIC_TRACE
+	cooldown_timer = HUNTER_PSYCHIC_TRACE_COOLDOWN
```

Comment:
same

Reply:
This is referenced elsewhere per the Q_DEL timer for the Hunter Mark tracker arrow; the others I agree with